lib/main.dart

```diff
@@ -1,28 +1,98 @@
+import "dart:async";
+import "dart:io";
+import "package:device_info_plus/device_info_plus.dart";
 import "package:flutter/material.dart";
+import "package:flutter_device_identifier/flutter_device_identifier.dart";
 import "package:flutter_dotenv/flutter_dotenv.dart";
 import "package:openapi_generator_annotations/openapi_generator_annotations.dart";
+import "package:oss_surveys_api/oss_surveys_api.dart";
 import "package:oss_surveys_customer/api/api_factory.dart";
+import "package:oss_surveys_customer/database/dao/keys_dao.dart";
+import "package:oss_surveys_customer/mqtt/listeners/surveys_listener.dart";
 import "package:oss_surveys_customer/mqtt/mqtt_client.dart";
 import "package:oss_surveys_customer/screens/default_screen.dart";
+import "package:oss_surveys_customer/theme/font.dart";
 import "package:oss_surveys_customer/theme/theme.dart";
 import "package:simple_logger/simple_logger.dart";
 
 final logger = SimpleLogger();
 final apiFactory = ApiFactory();
 
+late final String environment;
+late bool isDeviceApproved;
+late final String deviceSerialNumber;
+
 void main() async {
   _configureLogger();
   await dotenv.load(fileName: ".env");
-  mqttClient.connect();
+  environment = dotenv.env["ENVIRONMENT"]!;
+  mqttClient.connect().then((_) => _setupMqttListeners());
+  deviceSerialNumber = await _getDeviceSerialNumber();
+  await loadOfflinedFont();
+  isDeviceApproved = await keysDao.isDeviceApproved();
+  _setupTimers();
   runApp(const MyApp());
 }
 
 /// Configures logger to use [logLevel] and formats log messages to be cleaner than by default.
 void _configureLogger({logLevel = Level.INFO}) {
-  SimpleLogger().setLevel(logLevel);
+  SimpleLogger().setLevel(logLevel, includeCallerInfo: true);
   SimpleLogger().formatter = ((info) => "[${info.time}] -- ${info.callerFrame ?? "NO CALLER INFO"} - ${info.message}");
 }
 
+/// Setups MQTT Listeners
+void _setupMqttListeners() {
+  SurveysListener();
+}
+
+/// Setups timers for background tasks ran on interval.
+void _setupTimers() async {
+  if (!isDeviceApproved) {
+    Timer.periodic(const Duration(seconds: 30), (timer) => _pollDeviceApprovalStatus(timer) );
```

Comment:
These kinds of things should be automatically corrected on file save via Dart formatter, are you using Flutter extension in VS Code and is the formatter enabled?
```suggestion
    Timer.periodic(const Duration(seconds: 30), (timer) => _pollDeviceApprovalStatus(timer));
```

Reply:
Resolved